hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -2430,6 +2432,71 @@ private class Emit(
           }
         }
 
+      case x@NDArraySlice(child, slicesIR) =>
```

Comment:
We need a check somewhere that `slicesIR.pType.nFields == child.pType.nDims`. I don't know what the current status of InferType vs InferPType is, but that would be the place, IMO.

Reply:
I don't see such a check currently.

Reply:
Added it in `InferType`